guides/2022-04-05-build-a-conduit-connector.mdx

```diff
@@ -0,0 +1,318 @@
+	destCfg := DestinationConfig{
+		APIKey:        cfg[DestinationConfigAPIKey],
+		ApplicationID: cfg[DestinationConfigApplicationID],
+		IndexName:     cfg[DestinationConfigIndexName],
+	}
+
+	if destCfg.APIKey == "" {
+		return fmt.Errorf("%q is a required parameter", DestinationConfigAPIKey)
+	}
+	if destCfg.ApplicationID == "" {
+		return fmt.Errorf("%q is a required parameter", DestinationConfigApplicationID)
+	}
+	if destCfg.IndexName == "" {
+		return fmt.Errorf("%q is a required parameter", DestinationConfigIndexName)
+	}
+
+	d.config = destCfg
+	return nil
+}
+```
+
+### Open
+
+`Open` is called after Configure to signal the plugin it can prepare to
+start producing records. If needed, the plugin should open connections in
+this function. 
+
+```go
+
+func (d *Destination) Open(ctx context.Context) error {
+	client := search.NewClient(d.config.ApplicationID, d.config.APIKey)
+	index := client.InitIndex(d.config.IndexName)
+	d.index = index
+	return nil
+}
+```
+
+### Write
+
+`Write` receives a Record and is supposed to write the record to the
+destination right away.
+```go
+
+func (d *Destination) Write(ctx context.Context, record sdk.Record) error {
+	 , err := d.index.SaveObject(Object(record))
+	if err != nil {
+		return fmt.Errorf("could not save object: %w", err)
+	}
+
+	sdk.Logger(ctx).Debug().
+		Int("taskId", res.TaskID).
+		Str("objectId", res.ObjectID).
+		Msg("saved object")
+
+	return nil
+}
+
+```
+
+The `Object` methods serializes the record. This is implemented in `object.go`.
+
+### Teardown
+
+`Teardown` signals to the plugin that there will be no more calls to any
+other function. 
+```go
+
+func (d *Destination) Teardown(ctx context.Context) error {
+	// do nothing
+	return nil
+}
+```
+
+## Building
+
+Now that we have all the needed methods implmented, we can build our new connector:
+
+```
+go build -o algolia cmd/algolia/main.go
+```
+
+This will compile our connector to binary.
+
+## Creating Pipeline with a new Connector
+
+To use your new connector, perform the following steps:
+
+1. Start Conduit.
+2. Create Pipeline: 
+
+```curl
+curl 
+```
+3. Create Source
+
+```json
+{
+	"type": "TYPE_SOURCE",
+	"plugin": "builtin:generator",
+    "config": {
+        "readTime": "5s",
+        "fields": "id:int",
+    }
+}
+```
+
+4. Create Destination
+
+```json
+{
+	"type": "TYPE_DESTINATION",
+	"plugin": "/path/to/algoila",
+    "config": {
+        "api_key": "",
+        "application_id": "",
+        "index_name": ""
+    }
+}
+```
+
+5. Start pipeline
+
+After, starting the pipeline, you should see the following in the Algolia:
+
+SCREENSHOT
+
+## What's Next
+
+In this guide, we have covered the basics of how to build a connector for Conduit. The best way to learn more about what's possible with Conduit is to check out the [Conduit documentation](https://docs.conduit.io/), the [Conduit API documentation](https://docs.conduit.io/api/), and the [Conduit SDK](/).
```

Comment:
https://docs.conduit.io/ still points to the main page instead of docs, maybe it would be better to change the link to https://www.conduit.io/docs/introduction/getting-started.